pkg/apis/pipeline/v1alpha1/task_validation_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+)
+
+var validSource = Source{
+	Name: "source",
+	Type: "git",
+}
+
+func TestTaskSpec_Validate(t *testing.T) {
+	type fields struct {
+		Inputs    *Inputs
+		Outputs   *Outputs
+		BuildSpec *buildv1alpha1.BuildSpec
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name:    "nil",
+			wantErr: true,
+		},
+		{
+			name: "valid inputs",
+			fields: fields{
+				Inputs: &Inputs{
+					Sources: []Source{validSource},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid outputs",
+			fields: fields{
+				Outputs: &Outputs{
+					Sources: []Source{validSource},
+				},
```

Comment:
i'm surprised we can create Tasks without `buildSpecs` - I guess we're assuming that will be caught by the serialization/deserialization?

Reply:
Yeah not sure, I'll play around with it.

Reply:
Done!